spectrocloud/data_source_ssh_key.go

```diff
@@ -0,0 +1,67 @@
+package spectrocloud
+
+import (
+	"context"
+	"encoding/base64"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/spectrocloud/palette-sdk-go/client"
+)
+
+func dataSourceSSHKey() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSSHKeyRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:          schema.TypeString,
+				Computed:      true,
+				Optional:      true,
+				ConflictsWith: []string{"name"},
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				Description: "Name of the ssh key. This name is used to identify the ssh key in the Spectro Cloud UI.",
+			},
+			"ssh_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Optional:  true,
+				Sensitive: true,
+				StateFunc: func(val any) string {
+					return base64.StdEncoding.EncodeToString([]byte(val.(string)))
+				},
+				Description: "The SSH key to be used for the cluster. This is the public key that will be used to access the cluster.",
+			},
+			"context": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "project",
+				ValidateFunc: validation.StringInSlice([]string{"", "project", "tenant"}, false),
+				Description:  "Context of the ssh key. This can be either project or tenant. If not specified, the default value is project.",
```

Comment:
```suggestion
				Description:  "Context of the ssh key. This can be either project or tenant. If not specified, the default value is `project`.",
```